tasks/download.sh

```diff
@@ -36,7 +36,18 @@ verify-file() {
 
 download() {
   printf '%s\n' "Downloading: ${1}"
-  curl -s -f -L -o "$2" "$1"
+  tmp_file=$(mktemp "peadm-download.XXX")
+  echo "Temporary file created at: ${tmp_file}"
+  download_file=$(curl -s -f -L -o ${tmp_file} "$1")
```

Comment:
The output of the curl command doesn't tell us if it succeeded or not. What we want to judge on is the exit code. There are three ways to do that, that I know of.

Option 1: && chaining
```
curl -s -f -L -o ${tmp_file} "$1" \
  && mv "${tmp_file}" "$2"
  && return 0

# Failure to return means something failed
echo "Error: Curl has failed to download the file"
echo "Removing temporary file: ${tmp_file}"
rm "${tmp_file}"
return 1
```

Option 2: use an if statement
```
if curl -s -f -L -o ${tmp_file} "$1"; then
  mv "${tmp_file}" "$2"
  return 0
else
  echo "Error: Curl has failed to download the file"
  echo "Removing temporary file: ${tmp_file}"
  rm "${tmp_file}"
  return 1
fi
```

Option 3: check/capture $?

```
curl -s -f -L -o ${tmp_file} "$1"
curl_exit_code="$?"
if [[ "$curl_exit_code" = 0 ]]; then
  mv "${tmp_file}" "$2"
  return 0
else
  echo "Error: Curl has failed to download the file; exit code ${curl_exit_code}"
  echo "Removing temporary file: ${tmp_file}"
  rm "${tmp_file}"
  return 1
fi
```

I think I would probably go with option 1 because of my familiarity with it in lots of scripts from my past, but whichever of these is most readable would work great. The key bit is that we care about curl's exit code, not about its output.

Reply:
Thanks for the suggestions @reidmv I went with option No.2, I find it more readable :)

I'm waiting for the [Install Github Action ](https://github.com/puppetlabs/puppetlabs-peadm/actions/runs/1283610934) to finish